tests/testing_sf_masked_utilities.py

```diff
@@ -0,0 +1,125 @@
+import os
+from collections.abc import Callable
+from dataclasses import dataclass, field
+from functools import cache
+
+import pandas as pd
+import pytest
+
+import pydough
+from pydough.database_connectors import (
+    DatabaseContext,
+    load_database_context,
+)
+from pydough.metadata import GraphMetadata
+from tests.testing_utilities import (
+    PyDoughPandasTest,
+    graph_fetcher,
+)
+
+
+@dataclass
+class PyDoughSnowflakeMaskedTest(PyDoughPandasTest):
```

Comment:
Inherit from original test. Only difference is how the refsol are passed.
In this case, we pass a dictionary of 3 dataframes (or None) one for each account type